Explicitly break Python < 3.6 support in all our Python libraries

gsnedders opened the pull request:
#### 5200ed991d0d5b434ad1fabdf39710dbf3ea8d1e
<pre>
Explicitly break Python &lt; 3.6 support in all our Python libraries
<a href="https://bugs.webkit.org/show_bug.cgi?id=260922">https://bugs.webkit.org/show_bug.cgi?id=260922</a>

Reviewed by Jonathan Bedard.

279289@main already added a check in webkitcorepy for Python 2, which is
effectively inherited by most of the rest of our libraries. It did,
however, put it after a number of other imports, which makes it likely
some unrelated import/syntax error will be raised before it is reached.

This won&apos;t necessarily always result in an ImportError, because if we
start using unsupported syntax in __init__.py these will instead fail to
import with SyntaxError.

* Tools/Scripts/libraries/reporelaypy/reporelaypy/__init__.py:
* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/__init__.py:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/__init__.py:
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/__init__.py:
* Tools/Scripts/libraries/webkitflaskpy/webkitflaskpy/__init__.py:
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py:
* Tools/Scripts/test-webkitpy:
* Tools/Scripts/webkitpy/__init__.py:

Canonical link: <a href="https://commits.webkit.org/279482@main">https://commits.webkit.org/279482@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fe216ce7822686ebee8d9103228fd101876cc6d8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/53588 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/32947 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/6096 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56872 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/4318 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/55894 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/40421 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/4121 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/43441 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/60/builds/2829 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/55686 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/31169 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/46314 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/24576 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/53432 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/28015 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/3652 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/2470 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/49753 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3807 "Build is in progress. Recent messages:OS: Ventura (13.6.6), Xcode: 14.3; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running run-layout-tests-in-stress-mode; Running layout-tests; Running re-run-layout-tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/58465 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/28750 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/3858 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/50844 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/53595 "Passed tests") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/29954 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/46486 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/50183 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/30885 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7906 "Built successfully and passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/29726 "Build is in progress. Recent messages:OS: Sonoma (14.4.1), Xcode: 15.3; Running apply-patch; Checked out pull request; Running compile-webkit") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/libraries/resultsdbpy/resultsdbpy/__init__.py:
```diff
@@ -24,6 +24,9 @@
 
+if sys.version_info < (3, 6):
+    raise ImportError("resultsdbpy requires Python 3.6 or above")
+
```
We shouldn't need this, `resultsdbpy` has never supported Python 2 (and I think requires `3.7` or `3.8`?)

gsnedders replied:
To be fair, it doesn't seem harmful to have this here—IIRC, from memory, the tests all pass with Python 3.6, but we can of course bump this higher (but I was starting with being as conservative as possible!).

JonWBedard commented on Tools/Scripts/libraries/reporelaypy/reporelaypy/__init__.py:
```diff
@@ -23,6 +23,9 @@
 import os
 import sys
 
+if sys.version_info < (3, 6):
```
We shouldn't need this, `reporelaypy` has never supported Python 2

gsnedders replied:
To be fair, it doesn't seem harmful to have this here—IIRC, from memory, the tests all pass with Python 3.6, but we can of course bump this higher (but I was starting with being as conservative as possible!).

gsnedders committed:
Explicitly break Python < 3.6 support in all our Python libraries
https://bugs.webkit.org/show_bug.cgi?id=260922

Reviewed by Jonathan Bedard.

279289@main already added a check in webkitcorepy for Python 2, which is
effectively inherited by most of the rest of our libraries. It did,
however, put it after a number of other imports, which makes it likely
some unrelated import/syntax error will be raised before it is reached.

This won't necessarily always result in an ImportError, because if we
start using unsupported syntax in __init__.py these will instead fail to
import with SyntaxError.

* Tools/Scripts/libraries/reporelaypy/reporelaypy/__init__.py:
* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/__init__.py:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/__init__.py:
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/__init__.py:
* Tools/Scripts/libraries/webkitflaskpy/webkitflaskpy/__init__.py:
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py:
* Tools/Scripts/test-webkitpy:
* Tools/Scripts/webkitpy/__init__.py:

Canonical link: https://commits.webkit.org/279482@main
